lib/LightPanels/LightPanels.js

```diff
@@ -524,30 +548,22 @@ module.exports = kind(
 	* @private
 	*/
 	transitionFinished: function (sender, ev) {
-		if (ev.originator === this._currentPanel) {
-			if ((this._indexDirection < 0 && this.popOnBack && this.index < this.getPanels().length - 1) ||
-				(this._indexDirection > 0 && this.popOnForward && this.index > 0)) {
+		var panel = ev.originator;
+
+		if (panel === this._currentPanel) {
+			if ((this._indexDirection < 0 && (this.popOnBack || this.cacheViews) && this.index < this.getPanels().length - 1) ||
+				(this._indexDirection > 0 && (this.popOnForward || this.cacheViews) && this.index > 0)) {
 				this.popPanels(this.index, this._indexDirection);
 			}
-			if (this._currentPanel.postTransition) {
-				asyncMethod(this, function () {
-					this._currentPanel.postTransition();
-				});
-			}
-			if (this._garbagePanels && this._garbagePanels.length) {
-				this.finalizePurge();
-			}
+			if (this.popQueue && this.popQueue.length) this.finalizePurge();
+			this.transitioning = false;
+		}
 
-			if (this._currentPanel) {
-				this._currentPanel.removeClass('transitioning');
-				this._currentPanel.activated && this._currentPanel.activated();
-			}
+		panel.removeClass('transitioning');
 
-			if (this._previousPanel) {
-				this._previousPanel.removeClass('transitioning');
-				this._previousPanel.deactivated && this._previousPanel.deactivated();
-			}
-		}
+		asyncMethod(function () {
+			panel.postTransition();
```

Comment:
For posterity + reference, a mild reworking to utilize states instead of events:

https://github.com/enyojs/enyo/compare/ENYO-1981-aarontam...ENYO-1981-states-aarontam
https://github.com/enyojs/moonstone/compare/ENYO-1981-aarontam...ENYO-1981-states-aarontam